parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -175,12 +188,53 @@ pub mod pallet {
 	pub(super) type ExecutionHeaders<T: Config> =
 		StorageMap<_, Identity, H256, ExecutionHeader, OptionQuery>;
 
+	/// Execution headers ring buffer map implementation
+
+	/// Index storage for execution header ring buffer map
+	#[pallet::storage]
+	pub(crate) type ExecutionHeaderBufferIndex<T: Config> = StorageValue<_, u64, ValueQuery>;
+
+	/// Intermediate storage for execution header mapping
+	#[pallet::storage]
+	pub(crate) type ExecutionHeaderMapping<T: Config> =
+		StorageMap<_, Identity, u64, H256, ValueQuery>;
+
+	/// Ring buffer Map for Execution header
+	pub(crate) type ExecutionHeaderRingBufferMap<T> = RingBufferMapImpl<
+		u64,
+		<T as Config>::ExecutionHeadersPruneThreshold,
+		ExecutionHeaderBufferIndex<T>,
+		ExecutionHeaderMapping<T>,
+		ExecutionHeaders<T>,
+		OptionQuery,
+	>;
+
 	/// Current sync committee corresponding to the active header.
-	/// TODO  prune older sync committees than xxx
 	#[pallet::storage]
 	pub(super) type SyncCommittees<T: Config> =
 		StorageMap<_, Identity, u64, SyncCommitteeOf<T>, ValueQuery>;
 
+	/// Sync committee ring buffer map implementation
+
+	/// Index storage for sync committee ring buffer
+	#[pallet::storage]
+	pub(crate) type SyncCommitteesBufferIndex<T: Config> = StorageValue<_, u64, ValueQuery>;
+
+	/// Intermediate storage for sync committee mapping
+	#[pallet::storage]
+	pub(crate) type SyncCommitteesMapping<T: Config> =
+		StorageMap<_, Identity, u64, u64, ValueQuery>;
+
+	/// Ring buffer Map for Sync committee
+	pub(crate) type SyncCommitteesRingBufferMap<T> = RingBufferMapImpl<
```

Comment:
Seems `StorageMap` and `StorageDoubleMap` already supports ring buffers. 

See for example: https://github.com/Snowfork/cumulus/blob/c9cca565af5f165db5fbea9a149f9594c9a76b6e/bridges/modules/parachains/src/lib.rs#L268

Reply:
No. It is just indication of Max number of items that are supposed to be in that map. They do not implement `RingBuffer`.

Reply:
Oops, my mistake. 🙏